### PR TITLE
Clean up index.qhp

### PR DESCRIPTION
**Pull request by couet**
remove white spaces in index.qhp to make `qhelpgenerator` work.

https://github.com/root-project/root/issues/8781

### CODE REVIEW

**Comment by ferdymercury** [comment]
Thanks for the PR! we need to rerun by hand qhelpgenerator after that line, because otherwise it's only run within the doxygen process that already ended.

File: documentation/doxygen/Makefile [hunk 1]
```diff
@@ -53,6 +53,7 @@ doxygen: filter pyzdoc
 	./makeinput.sh
 	doxygen
 	./listlibs.sh
+	sed -i -e '/^ */s///' -e '/ *$$/s///'  $(DOXYGEN_OUTPUT_DIRECTORY)/html/index.qhp
```
Comment (ferdymercury) [inline documentation/doxygen/Makefile:56]: qhelpgenerator $(DOXYGEN_OUTPUT_DIRECTORY)/html/index.qhp

Comment (couet) [reply documentation/doxygen/Makefile:56]: done.

**Comment by couet** [comment]
Can you approuve if you think it is ok ?

File: documentation/doxygen/Makefile (hunk 1 above)
Comment (ferdymercury) [reply documentation/doxygen/Makefile:56]: Looks good now! I do not know if we will have to 'mv' the output file then by hand to the doxygen output directory, or if it generates it directly there.
Probably we also need to rename it from index.qch to ROOT.qch, (which doxygen does behind the scenes after running qhelpgenerator)

Comment (couet) [reply documentation/doxygen/Makefile:56]: index.qch is already in the right place. No need to move it. 
Ah yes will rename it also.

Comment (couet) [reply documentation/doxygen/Makefile:56]: > index.qch to ROOT.qch

you mean `.qhp` I guess ?

**Comment by couet** [comment]
If it is ok now for you feel free to approuve it.

File: documentation/doxygen/Makefile (hunk 1 above)
Comment (ferdymercury) [reply documentation/doxygen/Makefile:56]: qhelpgenerator transforms a qhp input file into a qch output file. I guess either we rename the input file, or the output file, both strategies should work afaik.

Comment (couet) [reply documentation/doxygen/Makefile:56]: Oh ok ... does the `.qch` disappear ?

**Review by ferdymercury — APPROVED** [review]
LGTM thanks again :)

File: documentation/doxygen/Makefile (hunk 1 above)
Comment (ferdymercury) [reply documentation/doxygen/Makefile:56]: Yes, the qhp is only an auxiliary file, so to say the intermediary file between doxygen and qt. qthelpgenerator transforms it to "binary" so to say. I think it does not disappear by itself, though we could remove it to save space if needed.

**Review by ferdymercury — APPROVED** [review]

File: documentation/doxygen/Makefile (hunk 1 above)
Comment (couet) [reply documentation/doxygen/Makefile:56]: Ok I rename the the `.qch` now

**Comment by ferdymercury** [comment]
This PR is probably no longer needed, as doxygen was very quick to fix the bug: https://github.com/doxygen/doxygen/pull/9068

I would suggest to update doxygen and trigger a rebuild to see if it runs well, before closing.

**Comment by couet** [comment]
I have updated doyen to the latest master version on the build machine. Let's see...

**Comment by ferdymercury** [comment]
Thanks! It looks as if with the new fix, they introduced a new problem https://lcgapp-services.cern.ch/root-jenkins/view/ROOT/job/root-makedoc-master/lastBuild/consoleText:

```
Error in line 1: Opening and ending tag mismatch.
error: failed to run qhelpgenerator on index.qhp
```

Could you share again the qhp file with me so that I can take a look ? Thanks!

**Comment by couet** [comment]
yes I am trying .That's always a challenge

**Comment by couet** [comment]
[index.qhp.gz](https://github.com/root-project/root/files/7941269/index.qhp.gz)

**Comment by ferdymercury** [comment]
Thanks! They are looking into it: https://github.com/doxygen/doxygen/issues/9058#issuecomment-1022147620

**Comment by ferdymercury** [comment]
This PR can be closed, as the removal of spaces is already done by doxygen in its master branch. (We still have the issue of mismatch between "opening" and "closing" braces, but that's another story that we can not easily solve by hacking the output).

**Comment by couet** [comment]
This PR is not needed anymore

**Pull request closed without merging**